com.adsbynimbus.nimbus/Runtime/Scripts/Nimbus.Internal/Android.cs

```diff
@@ -97,9 +108,19 @@ internal override void ShowAd(NimbusAdUnit nimbusAdUnit) {
 				holdTime = 5;
 				if (nimbusAdUnit.AdType == AdUnitType.Rewarded) holdTime = (int)TimeSpan.FromMinutes(60).TotalSeconds;
 			}
-
-			_helper.CallStatic(functionCall, _currentActivity, nimbusAdUnit.RawBidResponse, shouldBlock, holdTime,
-				listener);
+			Debug.unityLogger.Log("SHOWING AD");
+			var mintegralAdUnitId = "";
+			var mintegralAdUnitPlacementId = "";
+			foreach (ThirdPartyAdUnit adUnit in mintegralAdUnits)
+			{
+				if (adUnit.AdUnitType == nimbusAdUnit.AdType)
```

Comment:
Are there any .Net or C# provided find methods that could replace this foreach loop?